front/lib/api/assistant/actions/github.ts

```diff
@@ -0,0 +1,415 @@
+        created: Date.now(),
+        configurationId: agentConfiguration.sId,
+        messageId: agentMessage.sId,
+        error: {
+          code: "github_configuration_error",
+          message: `Github actions have not been configured for this workspace.`,
+        },
+      };
+      return;
+    }
+
+    if (
+      !rawInputs.owner ||
+      !rawInputs.repo ||
+      !rawInputs.pullNumber ||
+      typeof rawInputs.owner !== "string" ||
+      typeof rawInputs.repo !== "string" ||
+      typeof rawInputs.pullNumber !== "number"
+    ) {
+      yield {
+        type: "github_get_pull_request_error",
+        created: Date.now(),
+        configurationId: agentConfiguration.sId,
+        messageId: agentMessage.sId,
+        error: {
+          code: "github_get_pull_request_parameters_generation_error",
+          message: `Error generating parameters for github get pull request action.`,
+        },
+      };
+      return;
+    }
+
+    const pullNumber = rawInputs.pullNumber as number;
+    const owner = rawInputs.owner as string;
+    const repo = rawInputs.repo as string;
+
+    // Create the action in database and yield an event
+    const action = await AgentGithubGetPullRequestAction.create({
+      owner,
+      repo,
+      pullNumber,
+      functionCallId,
+      functionCallName: actionConfiguration.name,
+      agentMessageId: agentMessage.agentMessageId,
+      step,
+    });
+
+    yield {
+      type: "github_get_pull_request_params",
+      created: Date.now(),
+      configurationId: agentConfiguration.sId,
+      messageId: agentMessage.sId,
+      action: new GithubGetPullRequestAction({
+        id: action.id,
+        params: {
+          owner,
+          repo,
+          pullNumber,
+        },
+        pullBody: null,
+        pullCommits: null,
+        pullDiff: null,
+        functionCallId,
+        functionCallName: actionConfiguration.name,
+        agentMessageId: agentMessage.agentMessageId,
+        step,
+      }),
+    };
+
+    const tokRes = await getOAuthConnectionAccessToken({
+      config: apiConfig.getOAuthAPIConfig(),
+      logger,
+      provider: "github",
+      connectionId: platformActionsConfiguration.connectionId,
+    });
+
+    if (tokRes.isErr()) {
+      yield {
+        type: "github_get_pull_request_error",
+        created: Date.now(),
+        configurationId: agentConfiguration.sId,
+        messageId: agentMessage.sId,
+        error: {
+          code: "github_connection_error",
+          message: `Error getting connection token for github: ${tokRes.error.message}`,
+        },
+      };
+      return;
+    }
+
+    const octokit = new Octokit({ auth: tokRes.value.access_token });
+
+    try {
+      const query = `
+      query($owner: String!, $repo: String!, $pullNumber: Int!) {
+        repository(owner: $owner, name: $repo) {
+          pullRequest(number: $pullNumber) {
+            body
+            commits(last: 32) {
```

Comment:
nits:
- would move the octokit interactions to a different file to not pollute the action code with those big graphql queries
- would have "GITHUB_GET_PULL_REQUEST_ACTION_MAX_COMMITS" as a clear constant

Reply:
Will look into factoring in upcoming PRs, but I'm not a big fan of spreading files around while behind feature flags. Will add a varialbe for COMMITs